nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java

```diff
@@ -257,17 +215,93 @@ private static String getDelimiter() {
 
     private static Pattern getAttributeRegex(ProcessContext context, FlowFile flowFile) {
         return Pattern.compile(
-                context.getProperty(ATTRIBUTE_REGEX).evaluateAttributeExpressions(flowFile).getValue()
+                context.getProperty(ATTRIBUTE_PATTERN).evaluateAttributeExpressions(flowFile).getValue()
+        );
+    }
+
+    enum FilterMode implements DescribedValue {
+        RETAIN(
+                "Retain",
+                "Retains only the attributes matching the filter, all other attributes are removed."
+        ),
+        REMOVE(
+                "Remove",
+                "Removes the attributes matching the filter, all other attributes are retained."
         );
+
+        private final String value;
+        private final String description;
+
+        FilterMode(final String value, final String description) {
+            this.value = value;
+            this.description = description;
+        }
+
+        @Override
+        public String getValue() {
+            return this.value;
```

Comment:
Instead of returning `value`, this can return `name()` so that the enum value matches the configured property value. That allows FilterMode.valueOf() to work directly with the configured property value.